transaction/src/manifest/generator.rs

```diff
@@ -1517,11 +1708,188 @@ where
+        }
+        GeneratorErrorKind::InvalidExpression(string) => {
+            let title = format!("invalid expression '{}'", string);
+            (title, "invalid expression".to_string())
+        }
+        GeneratorErrorKind::InvalidBlobHash { actual, err } => {
+            let title = format!("invalid blob hash '{}' - {}", actual, err);
+            (title, "invalid blob hash".to_string())
+        }
+        GeneratorErrorKind::BlobNotFound(string) => {
+            let title = format!("blob with hash '{}' not found", string);
+            (title, "blob not found".to_string())
+        }
+        GeneratorErrorKind::InvalidBytesHex(string) => {
+            let title = format!("invalid hex value '{}'", string);
+            (title, "invalid hex value".to_string())
+        }
+        GeneratorErrorKind::NameResolverError(error) => match error {
+            NameResolverError::UndefinedBucket(string) => {
+                let title = format!("undefined bucket '{}'", string);
+                (title, "undefined bucket".to_string())
+            }
+            NameResolverError::UndefinedProof(string) => {
+                let title = format!("undefined proof '{}'", string);
+                (title, "undefined proof".to_string())
+            }
+            NameResolverError::UndefinedAddressReservation(string) => {
+                let title = format!("undefined address reservation '{}'", string);
+                (title, "undefined address reservation".to_string())
+            }
+            NameResolverError::UndefinedNamedAddress(string) => {
+                let title = format!("undefined named address '{}'", string);
+                (title, "undefined named address".to_string())
+            }
+            NameResolverError::NamedAlreadyDefined(string) => {
+                let title = format!("name already defined '{}'", string);
+                (title, "name already defined".to_string())
+            }
+        },
+        GeneratorErrorKind::IdValidationError { err, name } => match err {
+            ManifestIdValidationError::BucketNotFound(bucket_id) => {
+                let title = if let Some(name) = name {
+                    format!("bucket '{}' not found", name)
+                } else {
+                    format!("bucket id '{:?}' not found", bucket_id)
+                };
+                (title, "bucket not found".to_string())
+            }
+            ManifestIdValidationError::ProofNotFound(proof_id) => {
+                let title = if let Some(name) = name {
+                    format!("proof '{}' not found", name)
+                } else {
+                    format!("proof id '{:?}' not found", proof_id)
+                };
+                (title, "proof not found".to_string())
+            }
+            ManifestIdValidationError::BucketLocked(bucket_id) => {
+                let title = if let Some(name) = name {
+                    format!("bucket '{}' locked", name)
```

Comment:
I might consider expanding this title to give more context, because "locked" isn't very clear to an end user e.g. "cannot consume bucket '{}' because it's believed to be currently locked. This may be caused by an existing proof created from the bucket."

Reply:
Done